### ISSUE TITLE
Syncing of Plex Libary fails / stalls

### ISSUE THREAD

**Issue by Claw256** [croneter/PlexKodiConnect#1196]
Hi,
I am trying to setup PlexKodiConnect on my Windows 10 PC, and the syncing of the library keeps failing, at the point of which the the on-screen messages reads "Getting TV Shows (Episodes)", and also gets stuck at 59% with said message. I have linked to my `kodi.log` in the form of a pastebin with my tokens removed at the bottom of this issue. The Plex server I am trying to sync belongs a friend of mine, and it has a quite big of a library (40,000 movies, and 12,000 TV shows), maybe that is what is causing the error? I'm not sure though.

I have wiped by Kodi AppData folder multiple times trying to get this to work, and every single time it gets stuck at the exact same spot, 59%.
I have tried clearing the database of Kodi and PKC within the Addon settings, but after I do that I get an error saying: `ValueError: invalid literal for int() with base 10`.

I noticed in the log it says right before the last item to sync at 59%, the debug log reads:

`DEBUG: PLEX.plex_companion: Client is no longer registered. Plex Companion still running on port 3005`

I'm also not sure that this means. If anyone could help me out with this, that would be great.

Link to log: https://pastebin.com/ArBXC1w9

Edit: Also this is on a fresh vanilla install of Kodi v18.7 Leia (32-bit).

**Comment by Claw256** [comment]
It's grand, I'd say I'll just use the Plex Addon

**Comment by bloqhed** [comment]
I have the same problem, I will gather debug logs the next time it occurs.  My libraries are decent sized.  ~3200 movies, ~980 tv series (27k episodes), 46k music tracks.

I can temporarily get around this by wiping out my kodi database, wipe out the PKC settings.  Set it back up, and first sync music only, then add TV to the synced libraries list, then finally add movies.  This lasts for a few days, somtimes more, before kodi shows it's syncing a random chunk of movies and stalls out, usually around 59% as well.

**Comment by croneter** [comment]
@bloqhed A DEBUG log file would be much appreciated! Thx

**Comment by croneter** [comment]
@Claw256 The last episode to sync - or the one that causes the crash, is this one:
```
NOTICE: PLEX.tvshows: ADD episode plex_id: 1158752 - Madeira, My Dear?
```
If you try multiple times to sync, will it always be that episode? If so, could you please upload the episodes metadata here on Github? Just drop the XML file here in this GitHub dialog while replying to my message; check the part `XML Information / XML File` for getting the XML file: https://support.plex.tv/articles/201998867-investigate-media-information-and-formats/

**Comment by croneter** [comment]
Unfortunately, this log entry has nothing to do with the bug at hand... 😞 
```
DEBUG: PLEX.plex_companion: Client is no longer registered. Plex Companion still running on port 3005
```

**Comment by airmarshall** [comment]
> @Claw256 The last episode to sync - or the one that causes the crash, is this one:
> 
> ```
> NOTICE: PLEX.tvshows: ADD episode plex_id: 1158752 - Madeira, My Dear?
> ```
> 
> If you try multiple times to sync, will it always be that episode? If so, could you please upload the episodes metadata here on Github? Just drop the XML file here in this GitHub dialog while replying to my message; check the part `XML Information / XML File` for getting the XML file: https://support.plex.tv/articles/201998867-investigate-media-information-and-formats/

I'm getting these base 10 errors as well.

I ran it twice to see if the episode error is the same, but my logs don't seem to listing the ADD Episode action.

[01_KODI_1.log](https://github.com/croneter/PlexKodiConnect/files/5213338/01_KODI_1.log)
[01_KODI_2.log](https://github.com/croneter/PlexKodiConnect/files/5213339/01_KODI_2.log)

I can logs or XML details you require to help rectify this.

Many Thanks

**Comment by croneter** [comment]
@airmarshall nice! Kindly share both the DEBUG log of a failed sync as well as the respective Plex metadata Xml files of the items that cause the sync to fail

**Comment by airmarshall** [comment]
@croneter Forgive me for being a n00b, can you help me procure a DEBUG log. The logs I pulled are from a clean install of libreElec with only PKC installed. I'd switched on debug logging in the GUI and the screen overlay was displayed. File list of the log zip attached.

![filelist](https://user-images.githubusercontent.com/8303563/93028740-b468db80-f60d-11ea-9212-dd55b6687659.PNG)

More background info, so this only happens with my TV library. If I only Sync Movies all is fine and working well. I'm using Direct Play with Option 1 Substitute Paths using PKC streaming from a local Unraid SMB share.

If I then include the TV library to the sync, I get the errors. If I then play a movie it all works fine until I stop the movie, at which point the movie frame freezes and the box becomes unresponsive. SSH to reboot or shutdown is the only way out at that point.

**Comment by croneter** [comment]
@airmarshall I only need the 01_KODI.log. Do delete your Plex tokens, see https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug. Just "drop" the file here when replying

**Comment by airmarshall** [comment]
@croneter isn't that what I've posted above? I put the _1 and _2 just to
differentiate between the two.

Or is the log level incorrect?

**Comment by croneter** [comment]
@airmarshall you shared a Screenshot. But you did not upload the log files themselves nor shared any links to them. I need the entire log files 😉

**Comment by croneter** [comment]
Ah yes, logs are good! Thanks! Me using my mobile 🙈 

Kindly share the Plex xml metadata of the items where PlexKodiConnect crashes. See here how to download them: https://support.plex.tv/articles/201998867-investigate-media-information-and-formats/

**Comment by airmarshall** [comment]
@croneter No worries. I can't identify the episode from the logs, it seems
to be failing on a TV series import, maybe the series called 11.22.63 do
you agree?

**Comment by croneter** [comment]
@airmarshall it concerns the item with Plex ID 795. So simply open the Plex metadata for ANY Plex item, then replace it's ID with 795 in the URL

**Comment by airmarshall** [comment]
@croneter Hopefully this is it. XML Attached as LOG file due unsupported extension
[795.log](https://github.com/croneter/PlexKodiConnect/files/5223472/795.log)
.

``

**Comment by croneter** [comment]
@airmarshall unfortunately, it doesn't seem to be the Plex item causing the crash. Could you please be so kind and look at the other tv show metadata (not seasons, not episodes) and search for the string `2680-255598096`?

**Comment by croneter** [comment]
@airmarshall it should NOT be one of these shows: 
```
2020-09-12 22:13:03.545 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 805 - A Young Doctor's Notebook
2020-09-12 22:13:03.547 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 826 - Water Life
2020-09-12 22:13:03.548 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1006 - An Idiot Abroad
2020-09-12 22:13:03.549 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 841 - American Dad!
2020-09-12 22:13:03.550 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1056 - Band of Brothers
2020-09-12 22:13:03.552 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1032 - Bad Education
2020-09-12 22:13:03.553 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1068 - Barry
2020-09-12 22:13:03.555 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1087 - Life
2020-09-12 22:13:03.555 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1110 - Bluestone 42
2020-09-12 22:13:03.556 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1099 - Blue Planet II
2020-09-12 22:13:03.557 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1135 - Bodyguard (2018)
2020-09-12 22:13:03.558 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1143 - Breaking Bad
2020-09-12 22:13:03.559 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1212 - Breeders
2020-09-12 22:13:03.560 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1226 - Brooklyn Nine-Nine
2020-09-12 22:13:03.561 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1359 - Chernobyl
2020-09-12 22:13:03.562 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1366 - Come Fly With Me
2020-09-12 22:13:03.562 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1408 - Doctor Foster
2020-09-12 22:13:03.563 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1378 - Dexter
2020-09-12 22:13:03.564 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1374 - D-Day: The Last Heroes
2020-09-12 22:13:03.565 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1421 - Downton Abbey
2020-09-12 22:13:03.566 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1442 - Episodes
2020-09-12 22:13:03.567 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1490 - Extras
2020-09-12 22:13:03.568 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1509 - Fargo
2020-09-12 22:13:03.569 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1543 - Fresh Meat
2020-09-12 22:13:03.570 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1624 - Frozen Planet
2020-09-12 22:13:03.571 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1580 - Friday Night Dinner
2020-09-12 22:13:03.572 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1633 - Game of Thrones
2020-09-12 22:13:03.574 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1719 - Gavin & Stacey
2020-09-12 22:13:03.575 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1742 - Homeland
2020-09-12 22:13:03.576 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1945 - I'm Alan Partridge
2020-09-12 22:13:03.576 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1960 - Morgan Spurlock: Inside Man
2020-09-12 22:13:03.577 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1849 - House of Cards (US)
2020-09-12 22:13:03.578 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1935 - Human Planet
2020-09-12 22:13:03.579 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 1994 - Killing Eve
2020-09-12 22:13:03.580 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2023 - Knowing Me, Knowing You with Alan Partridge
2020-09-12 22:13:03.580 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2031 - Line of Duty
2020-09-12 22:13:03.581 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2066 - Louis Theroux
2020-09-12 22:13:03.581 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2109 - Louis Theroux's LA Stories
2020-09-12 22:13:03.582 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2114 - Louis Theroux's Weird Weekends
2020-09-12 22:13:03.582 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2136 - Luther
2020-09-12 22:13:03.583 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2162 - Mad Men
2020-09-12 22:13:03.584 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2285 - Mid Morning Matters with Alan Partridge
2020-09-12 22:13:03.584 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2308 - Modern Family
2020-09-12 22:13:03.585 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2262 - Making a Murderer
2020-09-12 22:13:03.586 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2594 - Mr. Robot
2020-09-12 22:13:03.588 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2579 - Mortimer & Whitehouse: Gone Fishing
2020-09-12 22:13:03.588 T:140240043882240  NOTICE: PLEX.tvshows: ADD tvshow plex_id: 2645 - Narcos
```

**Comment by croneter** [comment]
Thanks a lot for reporting this! PKC version 2.12.3 should fix this issue. Please let me know here if this is not the case.

**Comment by airmarshall** [comment]
Meant to post this the other day....sorry.

Your list of shows it shouldn't be interestingly ends at Narcos, that is
where the progress bar gets to everytime it fails.

The next show alphabetically is Narcos Mexico. Looking at that show in plex
is an anomaly because there is no artwork or episode/show info. Refreshing
metadata doesn't do anything either. The show didn't have a positive match.

I checked the folder structure and file names. Narcos- Mexico.

I've now renamed to Narcos Mexico including all the file names. I did The
Plex Dance and then did Fix Match as it still didn't pick it up properly.

After all that PKC synced successfully.

I don't know whether it was the file naming or fact it didn't have a
positive match but hopefully this process has been useful!

Thanks for the active support and an amazing plugin!

**Comment by croneter** [comment]
Thanks for the feedback @airmarshall  ! Sounds like somehow Plex messed up that one TV show 🙈 . Glad you could sort it all out...

**Comment by Claw256** [comment]
Hi @croneter , I came back to PKC after a break. During the server sync, I'm getting the same error code, `ValueError: invalid literal for int() with base 10:`. I tried resetting the database and the addon, and that didn't work.

I can make another issue on GitHub if this is getting too crowded in the comments of this issue? Here's my Sanitised Log (Too big for pastebin I'm afraid): http://www.mediafire.com/file/hplcl0egfifkspu/kodi.log

I read your above message about the episode `Madeira, My Dear`, should I get the XML of that file again? I'm not the owner of the Plex server, how would I go about getting that XML file?

I could also be getting timed out from the requests, as they is a lot of episodes (Around 300,000) of TV shows (Around 10,000), I don't know.
